modules/applications/pages/ui-builder/how-to-interact-with-tasks.adoc

```diff
@@ -37,69 +60,91 @@ For example:
 
-[source,JS]
+image:ui-builder/guides/create_modal.png[create_modal]
+
+Here, we will create a modal that will execute the "Review and answer claim" task from the xref:getting-started:create-web-user-interfaces.adoc[Getting Started] tutorial. The modal should look like this:
+
+image:ui-builder/guides/modal_task.png[modal_task]
+
+Go back to the `Queries` tab and create the following query:
+
+* `Name`: executeReviewAndAnswerClaim
+* `Method`: POST
+* `URL`: /bonita/API/bpm/userTask/{{Table1.triggeredRow.id}}/execution
+* `Body`:
+[source, JSON]
 ----
-export default {
-  submitCurrentTask(){
-    var task;
-    switch(this.currentTask.name){
-      case "Assign Workspace":
-        // this is the 'POST /execution' query shown in the previous section
-        task = executeAssignWorkspaceTask;
-        break;
-      case "Another task":
-        //...
-	break;
+{{
+  {
+    claimInput: {
+      answer: Input1.text
     }
-    task.run({taskId: this.currentTask.id}).then(response => {
-      //show alert, etc.
-    }).catch((error) => {
-      showAlert('fail', 'error');
-    });
-  },
+  }
+}}
 ----
 
+The `claimInput` contract of the task expects a parameter called 'answer', which will be fetched from the `Input1` widget.
 
-=== Bind JavaScript objects to a form
+This query will use the task id fetched from the table (`Table1.triggeredRow.id`) to execute the task. 
 
-Once a JavaScript object is created, you can bind it to your form. 
+Then, open the modal and bind the `executeReviewAndAnswerClaim` query to the `Onclick` property of the confirm button:
 
-Here, when the submit button is clicked, the `submitCurrentTask` JavaScript object is called:
+image:ui-builder/guides/bind_confirm_modal.png[bind_confirm_modal]
 
-image:ui-builder/guides/submit_task_form.png[submit_task_form]
 
 
 [NOTE]
 ====
-You don't necessarily need JavaScript objects to execute tasks. You can just create execution API queries and simply bind them to any kind of widgets. The choice is up to you.
+If you have other tasks you would like to manage, you can create new modals and new execution queries in the same way.
+
+To make sure you open the right modal, you can add the following code in the `Onclick` property of the "Open task" column: `{{ currentRow.displayName === "my task name" ? showModal(Modal1.name) : currentRow.displayName === "my other task name" ? showModal(Modal2.name) : null }}`
 ====
+
+
+=== Retrieve business data 
+
+If you would like to display business data, context and information from your previous tasks, here is what you should do:
+
+* Call `GET /userTask/context` to retrieve a storageId. For example: `GET /bonita/API/bpm/userTask/{{Table1.triggeredRow.id}}/context` 
+* Call `GET /API/bdm/businessData` to retrieve your business data, using the storageId returned previously. For example: `GET /API/bdm/businessData/com.company.model.Claim?c=1000&p=0&q=findByPersistenceId&f=persistenceId={{storageId}}`
```

Comment:
**suggestion:** Try this to avoid `{{storageId}}` being interpreted as an asciidoc variable.
```suggestion
* Call `GET /API/bdm/businessData` to retrieve your business data, using the storageId returned previously. For example: `GET /API/bdm/businessData/com.company.model.Claim?c=1000&p=0&q=findByPersistenceId&f=persistenceId={{{storageId}}}`
```

Reply:
```suggestion
* Call `GET /API/bdm/businessData` to retrieve your business data, using the storageId returned previously. For example: `GET /API/bdm/businessData/com.company.model.Claim?c=1000&p=0&q=findByPersistenceId&f=persistenceId=+{{storageId}}+`
```

Reply:
@marcobonita @csouchet  fyi

In Asciidoctor, the syntax {{storageId}} is interpreted as an attribute because double curly braces are used to denote placeholders for attributes. This is a common feature in many templating languages and is used to dynamically insert values.  

To prevent Asciidoctor from interpreting {{storageId}} as an attribute, you can escape the curly braces. Here are a few ways to do this:  

**Escape with backslashes:**  
`\{\{storageId\}\}`

**Use the pass:[...] macro:**
`pass:[{{storageId}}]`

**Use the + delimiter:**  
`+{{storageId}}+`

These methods will ensure that the curly braces are treated as literal characters and not as attribute placeholders.

Reply:
https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#attributes-and-substitutions